crates/rpc/rpc-engine-api/src/engine_api.rs

```diff
@@ -50,10 +53,35 @@ impl<Client: HeaderProvider + BlockProvider + StateProvider> EngineApi<Client> {
             EngineApiMessage::GetPayload(payload_id, tx) => {
                 let _ = tx.send(self.get_payload(payload_id).ok_or(EngineApiError::PayloadUnknown));
             }
-            EngineApiMessage::NewPayload(payload, tx) => {
+            EngineApiMessage::GetPayloadBodiesByHash(hashes, tx) => {
+                let _ = tx.send(self.get_payload_bodies_by_hash(hashes));
+            }
+            EngineApiMessage::GetPayloadBodiesByRange(start, count, tx) => {
+                let _ = tx.send(self.get_payload_bodies_by_range(start, count));
+            }
+            EngineApiMessage::NewPayload(version, payload, tx) => {
+                if let Err(err) = self.validate_withdrawals_presence(
+                    version,
+                    payload.timestamp.as_u64(), // TODO: shanghai valid for v1? https://github.com/ethereum/go-ethereum/blob/6428663faf50f8368cedf0297063154483cce72b/eth/catalyst/api.go#L412-L414
```

Comment:
you seem to be handling this already below, no? shanghai/withdrawals are rejected for v1

Reply:
i left this comment because geth doesn't reject post shanghai payloads in v1, so i posted a msg in r&d
https://discord.com/channels/595666850260713488/892088344438255616/1076421926035128341